### PR TITLE
test: deploy scripts

### PR DESCRIPTION
**Pull request by ErinHales**
- asserts that all elements in a deploy are defined
 - asserts that the governor chain has governorship over all other chains
 - asserts that governance router has ownership over other contracts
 - asserts home is set on xAppConnectionManager

TODO:
 - [x] Check deployNChains works with 1 chain
 - [x] Add checkBridgeDeploy

### CODE REVIEW

**Comment by anna-carroll** [comment]
Looking good. Can we update our test suite to run against contracts deployed using the N deploy script?

File: typescript/optics-tests/test/deploy.test.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+const domains = [1000, 2000, 3000];
+
+function assertBeaconProxy(beaconProxy: BeaconProxy<types.Contract>) {
+  expect(beaconProxy.beacon).to.not.be.undefined;
+  expect(beaconProxy.proxy).to.not.be.undefined;
+  expect(beaconProxy.implementation).to.not.be.undefined;
+}
+
+export async function checkDeploy(
```
Comment (prestwich) [inline typescript/optics-tests/test/deploy.test.ts:20]: love this, we should have it in the `optics-deploy` package instead of in `optics-test`

Comment (prestwich) [reply typescript/optics-tests/test/deploy.test.ts:20]: basically I think this should be part of the official deploy process :)

Comment (prestwich) [reply typescript/optics-tests/test/deploy.test.ts:20]: the next steps:

- for an n-chain deploy, check that all chains have enrolled replicas for all other chains
- `checkBridgeDeploy`

**Comment by ErinHales** [comment]
> Looking good. needs conflicts resolved.
> 
> Can we update our test suite to run against contracts deployed using the N deploy script? => in cross-chain tests, replace `deployTwoChains` with `deployNChains`

Yes! Will update that and resolve conflicts

**Comment by ErinHales** [comment]
Update: couple tests failing.  Debugging now

**Comment by ErinHales** [comment]
I think the deploy is fine, just sent an extra message so we need to generate a new proof with the correct index.  I also don't want to do that tho...

**Comment by ErinHales** [comment]
> I think the deploy is fine, just sent an extra message so we need to generate a new proof with the correct index.  I also don't want to do that tho...

Scratch that. I think we may be doing a step twice which results in an extra message being sent

File: typescript/optics-deploy/src/test/core.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import { expect } from 'chai';
```
Comment (prestwich) [inline typescript/optics-deploy/src/test/core.ts:1]: I think I'd prefer this file to be at `src/core/checks.ts`

Comment (prestwich) [reply typescript/optics-deploy/src/test/core.ts:1]: that way we can later add `src/bridge/checks.ts` and it'll be ocnsistent

Comment (ErinHales) [reply typescript/optics-deploy/src/test/core.ts]: Much better, thanks

File: solidity/optics-xapps/contracts/bridge/test/TestBridgeRouter.sol [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.6.11;
+
+import {BridgeRouter} from "../BridgeRouter.sol";
+
+contract TestBridgeRouter is BridgeRouter {
+    function getRemoteRouter(uint32 _domain) external view returns (bytes32) {
```
Comment (prestwich) [inline solidity/optics-xapps/contracts/bridge/test/TestBridgeRouter.sol:7]: I opened #750 to replace this test contract

Comment (prestwich) [reply solidity/optics-xapps/contracts/bridge/test/TestBridgeRouter.sol:7]: Cool you can refactor to simplify now :)

**Review by prestwich — DISMISSED** [review]

**Comment by prestwich** [comment]
ran prettier. will merge when checks pass

**Comment by ErinHales** [comment]
Will resolve conflicts here in a minute

**Review by prestwich — DISMISSED** [review]

**Review by prestwich — APPROVED** [review]

**Pull request merged**